tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala

```diff
@@ -288,6 +288,21 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
     val pattern = Seq("a.")
     val inputs = Seq("abc", "a\n\rb", "a\n\u0085b", "a\u2029\u0085b", "a\u2082\rb")
     assertCpuGpuMatchesRegexpFind(pattern, inputs)
+
+  }
+
+  test("replace_replace - ?, *, +, and {0, n} repetitions") {
+    val patterns = Seq("D?", "D*", "D+", "D{0,}", "D{0,1}", "D{0,5}", "[1a-zA-Z]{0,}",
+        "[1a-zA-Z]{0,2}", "A+")
+    val inputs = Seq("SS", "DD", "SDSDSDS", "DDDD", "DDDDDD", "ABCDEFG",
+        "\u000e\u0011\u0006h¹TEST\ud720AA")
```

Comment:
This test fails in CI due to the use of the superscript character. I filed https://github.com/NVIDIA/spark-rapids/issues/5521 to investigate this. We do not use characters like this in other tests so I think fixing this is outside of the scope of this PR.

```suggestion
    val inputs = Seq("SS", "DD", "SDSDSDS", "DDDD", "DDDDDD", "ABCDEFG")
```